app/templates/orders/_orders_back_link_path.hbs

```diff
@@ -19,16 +19,11 @@
       <i class="back_icon" aria-hidden="true">{{fa-icon 'angle-left'}}</i>
       <div>{{t "back"}}</div>
     {{/link-to}}
-  {{else if (is-or filterService.getOrderStateFilters.length filterService.getOrderTypeFilters.length)}}
-      {{#link-to "orders.index" (query-params isFiltered=true) classNames="back" }}
-        <i class="back_icon" aria-hidden="true">{{fa-icon 'angle-left'}}</i>
-        <div>{{t "back"}}</div>
-      {{/link-to}}
   {{else}}
-    {{#link-to "orders.index" classNames="back" }}
+    {{#back-link}}
       <i class="back_icon" aria-hidden="true">{{fa-icon 'angle-left'}}</i>
       <div>{{t "back"}}</div>
-    {{/link-to}}
+    {{/back-link}}
```

Comment:
I couldn't bring myself to add one more conditional here. It's too bad a practice, so attempting something a bit different.

Changes:
- Removed the first `orders.index` link because `isFiltered=true` doesn't seem to be used anywhere. Probably some old code
- Used a back-link as a "default" behaviour

Questions: 

What are the cases in which we **don't** want to navigate back to where we came from. I think a history.back would actually handle a lot of the cases written above.
I think that starting by finding the exceptions, and leaving the rest to default is a good start. Rather than listing the actual expected navigations one by one.